components/Form/index.js

```diff
@@ -0,0 +1,171 @@
+import styles from "./Form.module.css";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+export const Form = () => {
+  const [formData, setFormData] = useState({
+    title: "",
+    size: "",
+    dueBy: "",
+    tasks: "",
+  });
+
+  const [projectsData, setProjectsData] = useState([]);
+  const [errors, setErrors] = useState({});
+
+  const router = useRouter();
+
+  useEffect(() => {
+    const storedProjects = localStorage.getItem("projectsData");
+    if (storedProjects) {
+      setProjectsData(JSON.parse(storedProjects));
+    }
+  }, []);
+
+  const validateForm = () => {
+    let newErrors = {};
+    let isValid = true;
+
+    // Title validation
+    if (!formData.title || formData.title.length > 20) {
+      newErrors.title = "Title is required and must be at most 20 characters";
+      isValid = false;
+    }
+
+    // Size validation
+    const validSizes = ["s", "m", "l", "xl"];
+    if (!validSizes.includes(formData.size.toLowerCase())) {
+      newErrors.size = "Size must be one of s, m, l, or xl";
+      isValid = false;
+    }
+
+    // Due by validation
+    const dueDate = new Date(formData.dueBy);
+    if (isNaN(dueDate) || dueDate < new Date()) {
+      newErrors.dueBy = "Due by must be a valid future date";
+      isValid = false;
+    }
+
+    // Tasks validation
+    if (!formData.tasks) {
+      newErrors.tasks = "Tasks are required";
+      isValid = false;
+    }
+
+    setErrors(newErrors);
+    return isValid;
+  };
+
+  const formSubmitHandler = (event) => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      const newProject = {
+        id: uuidv4(),
+        projectTitle: formData.title,
+        projectSize: formData.size,
+        projectDue: formData.dueBy,
+        tasks: formData.tasks.split(",").map((task) => task.trim()),
+      };
+
+      localStorage.setItem(
+        "projectsData",
+        JSON.stringify([...projectsData, newProject])
+      );
```

Comment:
This is something you could best put in a separate `useEffect` which reacts to a change in `projectData`.
After you call `setProjectData` the `useEffect` will notice that the state changed and run its callback.
```js
useEffect(() => {
    localStorage.setItem(
        "projectsData",
        JSON.stringify([...projectsData, newProject])
    );
}, [projectData]);
```

Reply:
From the perspective of React `localStorage` is an external system. `localStorage` not part of React but of the browser. _Whenever you want to synchronize your app's state with an external system, you use `useEffect`._

In your case, you manage project data (your state) which you want to save to `localStorage` (the browser). Saving to the browser's `localStorage` is a considered a side-effect. You can [read more here](https://react.dev/learn/synchronizing-with-effects).